### DISCUSSION TITLE
Best approach to model crowded cluster galaxies

### DISCUSSION THREAD

**Discussion by joaquinhernandezg** [PetroFit/petrofit#161] [General]
Hi, I write to ask what you think is the best approach to model cluster galaxies. I am interested in measuring sersic indexes and effective radius, and I am getting poor models when fitting galaxies with high sersic indexes (a ring-like structure at the center in the residuals). I am making stamps around each galaxy, calculating the segmentation image, then the parameter guesses using the Petrosian correction, and then fitting the stamp using a mask, and I suspect it could be an issue with the background subtraction and neighbors contamination, so I also tried adding a Const2D model to the fitting. I perform global and local background subtraction, and I tried several algorithms and combinations, and also tried single and multiple sersic profiles (for modeling bright sources in the stamps), changing the number of iterations in the fitting, applying weights (the inverse of the sqrt of the flux seems to produce better models in my case), but I can not make all fits converge to a nice model, although faint sources and sources with n<~3 are well modeled. Do you have a clue on what could be happening and how to tackle the problem? I am aware that this is not a simple problem, but any guidance would be helpful.

Here I attach some residuals from fits I did in individual stamps and then combined into a single image.
![Screen Shot 2022-09-07 at 12 36 44](https://user-images.githubusercontent.com/42193649/188932645-4bd20872-74ff-4864-b69d-f1a9d1ab146f.png)

**Comment by robelgeda** [comment]
Hi @joaquinhernandezg, I am very sorry about the late reply, I just noticed your post. I agree based on the the residual image that the fit is not correct or close. There are a couple of places where there might have been issues. Let me start by asking some questions:
1. I am willing to look at your data if it is public or if you are able to send it to me. I can setup an example notebook and send it to you to get things rolling (I am assuming this is the quickest way).
2. What are you using as your PSF? Can you please post an image of the PSF you are using? Is it centered with the PSF image having odd dimensions? Is the PSF normalized where if you sum the pixel values it equals one? The centering of the PSF is critical for high n sources. 
3. Are you using default bounds (see the `get_default_sersic_bounds`) function.
4. Are you fitting the whole image at once? Or are you fitting each galaxy one at a time? I recommend making cutouts of isolated sources and seeing if the issue is still there. I also do not recommend fitting everything at once for a cluster environment like this. 
5. How many components are you using for the dominant/central galaxies? The central galaxies are likely to be better fitted with multiple (concentric) Sersic profiles. 

Some of my own thoughts:
I think what is getting you here is your ellipticity initial guesses. I do not think having corrected radii is needed for an image like this, the fitter can converge without having high accuracy radii guesses. It would also help to have multiple components for the bight central galaxies. 

My suggestion would be to break the image down into cutouts and fit  individuals galaxies. Then transform the `x_0` and `y_0` values of the model back to the image coordinates. 

These are some high level suggestions but I could provide specific advice once I hear back from you. It would be quick to do this if I can see your image and PSF,  that way I can send you a Jupyter Notebook with an example fit of one of the high index galaxies. If this sounds good, you can email me at `robel -at- princeton.edu`.
